Bottom: 8ea8bb857314d10598c0e3b9b8013e1c4f72ba3d
Top:    fda2f9d0fb70b615f718eff44adc53ca61334a3f
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-22 17:59:14 +0000

mymem.c: Allow elements to exist independently of a buffer.

If the DataType pointer to GetElementMemory is NULL, we allocate
memory but don't add to the buffer's element list or counters.

This is useful as it will allow rendering of pinout previews without
needing separate (or static) memory allocation to hold the previewed
Element.


---

diff --git a/src/mymem.c b/src/mymem.c
index 3de31f6..b6b3c30 100644
--- a/src/mymem.c
+++ b/src/mymem.c
@@ -408,8 +408,12 @@ GetElementMemory (DataType *data)
   ElementType *new_obj;
 
   new_obj = g_slice_new0 (ElementType);
-  data->Element = g_list_append (data->Element, new_obj);
-  data->ElementN ++;
+
+  if (data != NULL)
+    {
+      data->Element = g_list_append (data->Element, new_obj);
+      data->ElementN ++;
+    }
 
   return new_obj;
 }
